posthog/storage/session_recording_v2_object_storage.py

```diff
@@ -0,0 +1,78 @@
+import abc
+import structlog
+from boto3 import client as boto3_client
+from botocore.client import Config
+from django.conf import settings
+
+logger = structlog.get_logger(__name__)
+
+
+class SessionRecordingV2ObjectStorageBase(metaclass=abc.ABCMeta):
```

Comment:
General comment that it feels like this whole object storage class is perhaps either over or under-engineered 😅 Having a generic one that is just some abstractions around the AWS SDK followed by a very concrete copy of _some_ of the functionality etc. just feels off. Not gonna block as I know there is a lot of testing going on here but I feel like if someone was writing all this fresh we would choose a different pattern (i.e. some helper methods that are given a client or something).